warehouse/email.py

```diff
@@ -0,0 +1,37 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyramid_mailer import get_mailer
+from pyramid_mailer.message import Message
+
+from warehouse import celery
+
+
+@celery.task(bind=True, ignore_result=True, acks_late=True)
```

Comment:
Lets consider this situation in different contexts, so that it's easy to decide whether to send multiple emails or no email in the event of a worker failure.

If we consider `acks_late=False`:

**1. Password reset email:**  After successful completion of reset request, we'll show a message to the user to verify their email for the process to be followed. Then they don't receive any email in the event of failure, it's fair that user can come back and request for reset again. May be this time we'll deliver the message and user will'll be happy :)

Here we've to consider, what if the email fails to deliver multiple times?
  
**2. User registration confirmation:**  May be it's fine if the email fails to deliver, but if we decide like it has to be delivered: Unlike password reset it's not fair that user can try register again, The only possibility is that re-queue the message(I think acks_late=True does the same).

**3. New author/maintainer added notification:** I think this is similar to registration confirmation.

If we consider `acks_late=True`:

In the event of worker failure, message will be re-queued for all the scenarios mentioned above and there is a possibility that messages will be delivered multiple times.

**My opinion:** 
1. If we fail to deliver the message, user might loose the trust on the system. 
2. If we deliver multiple emails, there are two possibilities.
  a) User may feel like it's spam.
  b) Since it's a known email, he/she may be fine with that.

**Note: ** I think we've to consider this "worker is not known to crash" :)